### PR TITLE
Create Shell Script and Blockchain Task

### PR DESCRIPTION
**Pull request by leader282**
This task needs the freshers to create a shell script using bash terminal that generates a report from a CSV file. All relevant resource links have been added and also demo CSV and its output are present. This will be a teaching task and so they have to make a ppt showing how they made the script, the commands used and the output.

### CODE REVIEW

**Comment by leader282** [comment]
Added the blockchain task too in this. Here the freshers need to create a ppt regarding public-private key encryption, digital signatures, wallets, etc.

File: Teaching/Blockchain/crypto.md
Comment (harshkhandeparkar) [inline Teaching/Blockchain/crypto.md:1]: Suggestion: Add a hands-on demonstration task.

File: Teaching/Blockchain/crypto.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+[Basics of Public-Private Key Encryption](https://skerritt.blog/how-does-public-key-cryptography-work/) \
+[RSA (advanced)](https://www.comparitech.com/blog/information-security/rsa-encryption/) \
+[Bitcoin Wallet](https://www.forbes.com/advisor/in/investing/cryptocurrency/what-is-a-bitcoin-wallet/#:~:text=Bitcoin%20wallets%20not%20only%20hold,can%20open%20your%20Bitcoin%20wallet/) \
+[More on Wallets](https://101blockchains.com/crypto-wallets/)
+
+## Note
+
+Try to focus more on understanding how things work rather than diving deep into the maths at this point. Diving into maths is good but not recommended for the current task at hand.
+
+This book can be used as a great reference : [Mastering Bitcoin](https://github.com/bitcoinbook/bitcoinbook/)
```
Comment (Yureien) [inline Teaching/Blockchain/crypto.md:37]: Add this reference: https://rosenbaum.se/book/grokking-bitcoin.html

File: Teaching/Blockchain/crypto.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+Research and gather information on the key concepts related to Public-Private Key Encryption and Bitcoin Wallets
+
+Create a presentation with the following content:
+- The basics of public-key cryptography and how it differs from traditional symmetric-key cryptography.
+- How public and private keys are generated and used to encrypt and decrypt data.
+- The concept of digital signatures and how they are used to verify the authenticity of data.
+- How public-private key encryption is used in Bitcoin wallets to secure and manage Bitcoin transactions.
+- The different types of Bitcoin wallets, including hot wallets and cold wallets, and the advantages and disadvantages of each.
+- Best practices for securing Bitcoin wallets, including the importance of backing up private keys and using multi-factor authentication.
```
Comment (Yureien) [inline Teaching/Blockchain/crypto.md:23]: Add another point:

- [Optional, but recommended] Use the `bitcoin-cli` client to make a regtest wallet, and try creating transactions.

Comment (Yureien) [reply Teaching/Blockchain/crypto.md:23]: Falls under hands-on demonstration.

**Review by Yureien — CHANGES_REQUESTED** [review]

**Pull request merged**